stm/stm.mli

```diff
@@ -110,6 +110,9 @@ val add : doc:doc -> ontop:Stateid.t -> ?newtip:Stateid.t ->
   bool -> Vernacexpr.vernac_control CAst.t ->
   doc * Stateid.t * [ `NewTip | `Unfocus of Stateid.t ]
 
+(* Returns the proof state before the most recent proof step.  Used to compute proof diffs *)
+val get_prev_proof : doc:doc -> Proof.t option
```

Comment:
I don't like the API to be aware of the current state, or at least, this is where ae are going with the entire API so the new code should not be stateful.
I think you should take in input the `Stateid.t` and use `Stm.get_current_state` at call site (and the comment needs to be fixed accordingly).